docs/DeveloperGuide.md

```diff
@@ -466,6 +466,16 @@ The following activity diagram summarises the flow of events when `DeleteQuestio
 
 Figure 5.2.3.2. Activity diagram for `DeleteQuestionCommand` execution
 
+##### 6.3.1.4 Design considerations
+
+The `Student` class guarantees immutability, including its list of `Question` objects. As such, any operation must return a modified copy instead of directly modifying the base list.
+
+The pros of doing so is that it becomes easier to test and debug as there are fewer mutations to worry about.
+
+The downside, however, that this approach is more memory-intensive and possibly slower, since changing a single question involves recreating the entire list of questions attached to the student.
+
+Having considered that our target audience (one-to-one private tutors) are unlikely to have so much data that this would severely impact performance, we believe that this is worth the trade-off.
```

Comment:
Does this contradict our NFR as one of them is "The program should be able to handle at least 100 students."